Strory7 add review form

ManpreetLamsal opened the pull request.

katame13 committed:
created review schema, entity, service, repo, and worked on the seeder and the tile

katame13 committed:
Merge branch 'main' into reviews

Christiandecolombia committed:
created review controller

Christiandecolombia committed:
Formatted controller file

katame13 committed:
worked on site descriptions

added V3.1 scheema
updated seeder, entity, and show page for descriptions
created empty react page for show
created api endpoint for show
added method to site service for optional find by id method

Christiandecolombia committed:
added NewReviewShowPage

Christiandecolombia committed:
Merge branch 'main' into reviews

katame13 committed:
having issues with displaying the show page

posting for help debuging

Christiandecolombia committed:
Added to models

Christiandecolombia committed:
Bugg fixing Bundle.js error

fbraese committed:
Merge branch 'Story3SiteShowPage' of https://github.com/katame13/RunTimeTerror into reviews
We need the code that was being worked on for reviews on the Story 3 branch yesterday

Christiandecolombia committed:
Finished review seeder

Christiandecolombia committed:
Merge branch 'main' into reviews

Christiandecolombia committed:
Finished Site show page with reviews. Completed review feature

Christiandecolombia committed:
Added to NewReviewForm

Christiandecolombia committed:
added to site show

Christiandecolombia committed:
finished review posting form

fbraese committed:
Merge branch 'main' of https://github.com/katame13/RunTimeTerror into Strory7AddReviewForm

fbraese committed:
Merge branch 'main' of https://github.com/katame13/RunTimeTerror into Strory7AddReviewForm
Updated from main

Christiandecolombia committed:
Created button to show form

Christiandecolombia committed:
Updated the front end validation.

ManpreetLamsal committed:
fixed errors

added @insertable= false to review entity.  Took out console logs and souts.

safang99 commented on src/main/frontend/components/NewReviewForm.js:
```diff
@@ -0,0 +1,96 @@
+    </form>
+  )
+}
+export default NewReviewForm
```
Content looks great! We can add empty lines between functions to help with readability.  For example between 34 & 35 and also 41&42.  I would skim through or use a formatter like `prettier` when saving

safang99 commented on src/main/frontend/components/SiteShow.js:
```diff
@@ -45,14 +76,38 @@ const SiteShow = (props) => {
-      <img src={imgUrl}/>
-        <a href={url}><p>Visit the Site</p></a>
+      <img src={imgUrl} />
+      <a href={url}><p>Visit the Site</p></a>
```
If this is an internal link we should use `<Link>` components instead of anchor tags, but if this is to an external page this is fine.  We may want to set the `target` attribute so it opens in a new tab though

safang99 commented on src/main/frontend/components/SitesIndex.js:
```diff
@@ -8,6 +8,7 @@ const SitesIndex = props => {
   const fetchSites = async () => {
     try {
       const response = await fetch("api/v1/sites")
+      console.log("inside index fetch")
```
log statement to remove

safang99 commented on src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java:
```diff
@@ -17,11 +26,36 @@
+
+
+    @PostMapping("/{id}/addreview")
+  public ResponseEntity createReview(@PathVariable Integer id, @RequestBody @Valid Review review,
```
shift 41 back so it lines up with 42

safang99 commented on src/main/frontend/components/SiteShow.js:
```diff
@@ -45,14 +76,38 @@ const SiteShow = (props) => {
+    />
+  }
+
+
```
extra empty line

safang99 commented on src/main/frontend/components/SiteShow.js:
```diff
@@ -31,11 +35,38 @@ const SiteShow = (props) => {
+  const postReviewForm = async (formPayload) => {
+    try {
+      const response = await fetch(
+        `/api/v1/reviews/${siteId}/addreview`, {
```
To follow a more restful pattern we can have this send it's POST request to `/api/v1/sites/{siteId}/reviews`.  This would be the way to nest a resource.  Then on the backend there are a few ways to handle this, but the easiest way forward would be to add a POST endpoint for `/{siteId}/reviews` to the `SitesController`

safang99 commented on src/main/java/com/launchacademy/reviews/controllers/ApiV1ReviewsController.java:
```diff
@@ -17,11 +26,36 @@
+      reviewData.put("review", review);
+      return new ResponseEntity<Object>(reviewData, HttpStatus.CREATED);
+    }
+  }
```
After updating the fetch path we can move this logic

safang99 reviewed (changes requested):
Looking good! We will want to update the fetch path being used to follow a more RESTful pattern and clean up a few formatting issues along the way

ManpreetLamsal committed:
made required changes

updated sites controller
deleted the reviewsController,
fixed formatting issues and added target to the external link.
